src/app/rundown-view/services/action-trigger-producer-key-binding.service.ts

```diff
@@ -185,7 +178,25 @@ export class ActionTriggerProducerKeyBindingService implements KeyBindingService
       return
     }
     this.rundown = rundown
-    this.emitNewKeybindings()
+    this.updateSystemKeyBindings()
+    this.emitKeybindings()
+  }
+
+  private updateSystemKeyBindings(): void {
+    if (!this.rundown) {
+      this.systemKeyBindings = []
+      return
+    }
+
+    if (this.rundown.mode === RundownMode.INACTIVE) {
+      this.systemKeyBindings = this.systemKeyBindingFactory.createInactiveRundownKeyBindings(this.rundown)
+      return
+    }
+
+    this.systemKeyBindings =
+      this.rundown.mode === RundownMode.ACTIVE
+        ? this.systemKeyBindingFactory.createActiveRundownKeyBindings(this.rundown)
+        : this.systemKeyBindingFactory.createRehearsalRundownKeyBindings(this.rundown)
```

Comment:
This seems like the perfect case for a switch statement :D
To reduce the complexity, the generation and assignment can be separated to only have one place where we assign. E.g.

```typescript
private updateSystemKeyBindings(): void {
  this.systemKeyBindings = this.createSystemKeyBindings()
}

private createSystemKeyBindings(): KeyBinding[] {
  if (!this.rundown) {
    return []
  }

  switch (this.rundown.mode) {
    case RundownMode.INACTIVE:
      return this.systemKeyBindingFactory.createInactiveRundownKeyBindings(this.rundown)
    case RundownMode.ACTIVE:
      return this.systemKeyBindingFactory.createActiveRundownKeyBindings(this.rundown)
    case RundownMode.REHEARSAL:
      return this.systemKeyBindingFactory.createRehearsalRundownKeyBindings(this.rundown)
  }
}

Reply:
Updated